doc_src/cmds/complete.rst

```diff
@@ -76,7 +76,7 @@ Command specific tab-completions in ``fish`` are based on the notion of options
 
 - Short options, like ``-a``. Short options are a single character long, are preceded by a single hyphen and can be grouped together (like ``-la``, which is equivalent to ``-l -a``). Option arguments may be specified by appending the option with the value (``-w32``), or, if ``--require-parameter`` is given, in the following parameter (``-w 32``).
 
-- Old-style long options, like ``-Wall`` or ``-name``. Old-style long options can be more than one character long, are preceded by a single hyphen and may not be grouped together. Option arguments are specified in the following parameter (``-ao null``) or after a ``=`` (``-ao=null``).
+- Old-style options, long like ``-Wall`` or ``-name`` or even short like ``-a``. Old-style options can be more than one character long, are preceded by a single hyphen and may not be grouped together. Option arguments are specified by default following a space (``-ao null``) or after a ``=`` (``-ao=null``). Short options for commands that do not support concatenating multiple short options in one string (not supporting ``-xf`` as short for ``-x -f``) or a short option and its value in one string (not supporting ``-d9`` instead of ``-d 9``) should be specified as single-character old-style options; for example, ``complete -c foo -o s; complete -c foo -o v`` would never suggest ``foo -ov`` but rather ``foo -o -v``.
```

Comment:
That's fine with me, but I would like to keep the omissions of "long" from the old-style documentation to make it more inclusive/so it isn't dismissed out-of-hand by someone thinking it can't apply to them in their situation.

fwiw, I don't think the existing completions are necessarily a good metric to go by. As shown in #9233, they may be wrong but just haven't annoyed anyone enough to complain. Option grouping is an old unix greybeard mainstay but there's no _requirement_ for it to be the case and anyone manually parsing arguments in a new application today instead of using getopts or a similar library is probably not handling them. As our completions library has expanded, we're no longer documenting just the most "elite" traditional cli tools, but also a lot of less traditional ones too. 

* Two out of the eleven surveyed C++ arg parsing libraries [in this 2018 roundup](https://attractivechaos.wordpress.com/2018/08/31/a-survey-of-argument-parsing-libraries-in-c-c/) didn't support option grouping
* Any cross-platform tools with Windows origins (like any .NET Core application) isn't likely to support option grouping because it's not a thing on Windows. 
* Commands not written in C/C++ and without a getopt legacy (such as any java applications) don't usually support option grouping either. 
* golang's most common arg parser [doesn't support grouping short options](https://nullprogram.com/blog/2020/08/01/#:~:text=Go%E2%80%99s%20flag%20package,with%20%3D.).
* Python's argparse [apparently broke short option grouping](https://utcc.utoronto.ca/~cks/space/blog/python/ArgparseSomeUnixNotes#:~:text=Changed%20in%20version%203.8%3A%20In%20previous%20versions%2C%20allow_abbrev%20%5Bbeing%20False%5D%20also%20disabled%20grouping%20of%20short%20flags%20such%20as%20%2Dvv%20to%20mean%20%2Dv%20%2Dv.) until Python 3.8.

Reply:
How about this wording:

> Old-style options can't be grouped together even if they are only a single character long, unlike short options.

And then, in the examples, show the example you gave here:

```fish
complete -c foo -o o
complete -c foo -o v
# This will allow "foo -o -v", but not "foo -ov" because old-style options can't be grouped
```

Reply:
>Two out of the eleven surveyed C++ arg parsing libraries [in this 2018 roundup](https://attractivechaos.wordpress.com/2018/08/31/a-survey-of-argument-parsing-libraries-in-c-c/) didn't support option grouping

(Both of those have since been fixed: https://tclap.sourceforge.net/v1.2/manual.html#COMBINE_SWITCHES and https://github.com/adishavit/argh/issues/28)

Yes, weird option parsers exist and we should, within reason, allow them to be supported with completions. I'm just saying that we don't need to front-load that complexity in the docs.